katharsis-core/src/main/java/io/katharsis/dispatcher/controller/resource/ResourcePatch.java

```diff
@@ -77,12 +78,22 @@ public BaseResponseContext handle(JsonPath jsonPath, QueryAdapter queryAdapter,
 
         String attributesFromFindOne = null;
         try {
-            // extract attributes from find one without any manipulation by query params (such as sparse fieldsets)
+            // extract current attributes from findOne without any manipulation by query params (such as sparse fieldsets)
             attributesFromFindOne = this.extractAttributesFromResourceAsJson(resource, jsonPath, new QueryParamsAdapter(new QueryParams()));
             Map<String,Object> attributesToUpdate = objectMapper.readValue(attributesFromFindOne, Map.class);
-            // get the JSON form the request and deserialize into a map
+            // deserialize the request JSON's attributes object into a map
             String attributesAsJson = objectMapper.writeValueAsString(dataBody.getAttributes());
             Map<String,Object> attributesFromRequest = objectMapper.readValue(attributesAsJson, Map.class);;
+
+            // remove attributes that were omitted in the request
+            Iterator<String> it = attributesToUpdate.keySet().iterator();
+            while(it.hasNext()) {
+            	String key = it.next();
+			    if(attributesFromRequest.containsKey(key))
```

Comment:
If you flip this predicate around you can simply get rid of the `continue` statement